livedata/src/main/java/io/getstream/chat/android/livedata/controller/QueryChannelsControllerImpl.kt

```diff
@@ -50,9 +51,8 @@ internal class QueryChannelsControllerImpl(
     override val endOfChannels: LiveData<Boolean> = _endOfChannels
 
     private val _channels = MutableLiveData<Map<String, Channel>>()
-    // Ensure we don't lose the sort in the channel
-    // TODO: The live list of channels doesn't actually maintain the specified query sort
-    override var channels: LiveData<List<Channel>> = Transformations.map(_channels) { cMap -> queryEntity.channelCids.mapNotNull { cMap[it] } }
+    // Keep the channel list locally sorted
+    override var channels: LiveData<List<Channel>> = Transformations.map(_channels) { cMap -> cMap.values.sortedWith(sort.channelComparator) }
```

Comment:
Have this change into another PR would be easier and faster to be reviewed

Reply:
I like to test the sample app thoroughly before merging a PR, bit too much overhead to split it up.